### PR TITLE
Proxy all requests - even a repeated one

### PR DESCRIPTION
**Pull request by sameena-ops**

### CODE REVIEW

File: src/WireMock.Net/Owin/WireMockMiddlewareOptions.cs [hunk 1]
```diff
@@ -95,4 +95,6 @@ internal class WireMockMiddlewareOptions : IWireMockMiddlewareOptions
 
     /// <inheritdoc />
     public QueryParameterMultipleValueSupport? QueryParameterMultipleValueSupport { get; set; }
+
+    public bool ProxyAll { get; set; }
```
Comment (StefH) [inline src/WireMock.Net/Owin/WireMockMiddlewareOptions.cs:99]: add `/// <inheritdoc />`

File: test/WireMock.Net.Tests/WireMockServer.Proxy.cs [hunk 2]
```diff
@@ -922,4 +922,49 @@ public async Task WireMockServer_Proxy_WhenTargetIsNotAvailable_Should_Return_Co
         server.LogEntries.Should().HaveCount(1);
         server.Stop();
     }
+
+    [Fact]
+    public async Task WireMockServer_ProxyAndRecordSettings_SameRequest_ShouldProxyAll()
+    {
+        //Arrange
+        WireMockServerSettings wireMockServerSettings = new WireMockServerSettings
```
Comment (StefH) [inline test/WireMock.Net.Tests/WireMockServer.Proxy.cs:930]: use var

File: test/WireMock.Net.Tests/WireMockServer.Proxy.cs [hunk 3]
```diff
@@ -922,4 +922,49 @@ public async Task WireMockServer_Proxy_WhenTargetIsNotAvailable_Should_Return_Co
...
+                Url = "http://postman-echo.com",
+                SaveMapping = true,
+                ProxyAll = true,
+                SaveMappingToFile = false,
+                ExcludedHeaders = new[] { "Postman-Token" },
+                ExcludedCookies = new[] { "sails.sid" }
+            }
+        };
+
+           WireMockServer server = WireMockServer.Start(wireMockServerSettings);
```
Comment (StefH) [inline test/WireMock.Net.Tests/WireMockServer.Proxy.cs:944]: Do a code-reformat

**Review by StefH — CHANGES_REQUESTED** [review]

File: src/WireMock.Net/Server/WireMockServer.Proxy.cs [hunk 4]
```diff
@@ -31,6 +31,10 @@ private void InitProxyAndRecord(WireMockServerSettings settings)
             proxyRespondProvider.AtPriority(WireMockConstants.ProxyPriority);
         }
 
+        if(settings.ProxyAndRecordSettings.ProxyAll){
+            proxyRespondProvider.AtPriority(int.MinValue);
```
[the remark this answers is not in the document]
Comment (sameena-ops) [reply src/WireMock.Net/Server/WireMockServer.Proxy.cs:36]: pushed the changes

File: src/WireMock.Net.Abstractions/Admin/Settings/ProxyAndRecordSettingsModel.cs [hunk 5]
```diff
@@ -73,4 +73,9 @@ public class ProxyAndRecordSettingsModel
     /// Defines the Replace Settings
     /// </summary>
     public ProxyUrlReplaceSettingsModel? ReplaceSettings { get; set; }
+
+    /// <summary>
+    /// Proxy all Api calls, irrespective of any condition
+    /// </summary>
+    public bool ProxyAll { get; set; } = false;
```
Comment (StefH) [inline src/WireMock.Net.Abstractions/Admin/Settings/ProxyAndRecordSettingsModel.cs:80]: This new field should also be parsed like it's done here:
WireMockServerSettingsParser.cs

private static void ParseProxyAndRecordSettings(WireMockServerSettings settings, SimpleSettingsParser parser)

**Review by StefH — APPROVED** [review]

**Comment by MindaugasLaganeckas** [comment]
Very nice feature! Thank you for working on it! Looking forward to get it released! 😄

**Comment by StefH** [comment]
It's released.

**Pull request merged**